Throw error when no characters specified

Lex-Ashu opened the pull request:
Added Error Handling in Password Generator.

Lex-Ashu committed:
Update __init__.py

Added Error Handling in Password Generator.

bsoyka commented on passwd/__init__.py:
```diff
@@ -182,7 +182,10 @@ def generate(
+        if not allowed_chars:
+            raise ValueError("No character types selected for password generation.")     #Error Handling
+        
+        return "".join(choice(allowed_chars) for _ in range(length))
```
This line is duplicated now—let's remove one of them.

bsoyka commented on passwd/__init__.py:
```diff
@@ -182,7 +182,10 @@ def generate(
         elif self.special:
             allowed_chars += punctuation
-
+        if not allowed_chars:
+            raise ValueError("No character types selected for password generation.")     #Error Handling
```
Can you add a test case for this in `tests/test_password_generator.py`?

Lex-Ashu replied:
I will resolve it till that can u mark it for hacktober accepted as it is urgent and I will surely update the code with your issues

bsoyka reviewed (changes requested):
Thanks for contributing! There are a couple of minor things to work on here—see my comments for more.

Lex-Ashu committed:
Update __init__.py

Removed the extra line

Lex-Ashu commented:
I have removed the duplicated line and I have added the required test case
in test_password_generator.py. Have a look at it and let me know if any
changes are needed or I would be grateful if you merge both of my pull
requests.
Thank You

On Mon, Oct 21, 2024 at 11:14 PM Ben Soyka ***@***.***> wrote:

> ***@***.**** requested changes on this pull request.
>
> Thanks for contributing! There are a couple of minor things to work on
> here—see my comments for more.
> ------------------------------
>
> In passwd/__init__.py
> <https://github.com/bsoyka/passwd/pull/42#discussion_r1809237631>:
>
> > @@ -182,7 +182,10 @@ def generate(
>              allowed_chars += punctuation if special else ""
>          elif self.special:
>              allowed_chars += punctuation
> -
> +        if not allowed_chars:
> +            raise ValueError("No character types selected for password generation.")     #Error Handling
> +
> +        return "".join(choice(allowed_chars) for _ in range(length))
>
> This line is duplicated now—let's remove one of them.
> ------------------------------
>
> In passwd/__init__.py
> <https://github.com/bsoyka/passwd/pull/42#discussion_r1809238992>:
>
> > +        if not allowed_chars:
> +            raise ValueError("No character types selected for password generation.")     #Error Handling
>
> Can you add a test case for this in tests/test_password_generator.py?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/bsoyka/passwd/pull/42#pullrequestreview-2382877359>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BLFSGBTQU6LSDUNUYCC3RJDZ4U4PRAVCNFSM6AAAAABP6KXCYOVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDGOBSHA3TOMZVHE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

bsoyka commented:
Both the main change and the tests need to be in the same pull request, so the test runs on the updated code and we can make sure everything's working properly.

Lex-Ashu committed:
Added test case

Lex-Ashu commented:
Done

bsoyka commented on tests/test_password_generator.py:
```diff
@@ -18,6 +18,9 @@ def test_length():
     assert len(all_0.generate()) == 0
 
+def test_no_character_types_selected():    #Added a test case when all character types are set to False
+    with raises(ValueError,match="No character types selected for password genration."):
```
You have a typo here (`genration`) which is causing the test to fail.

Lex-Ashu committed:
Done

bsoyka reviewed (approved):
Tests are passing now—looks good to me! Thanks for your contributions!
